src/dmd/statement.d

```diff
@@ -398,27 +398,35 @@ extern (C++) abstract class Statement : ASTNode
         v.visit(this);
     }
 
-  pure nothrow @nogc:
+    /************************************
+     * Does this statement end with a return statement?
+     * Returns:
+     *  return statement it ends with, otherwise null
+     */
+    pure nothrow @nogc
+    inout(ReturnStatement) endsWithReturnStatement() inout { return null; }
```

Comment:
IMO the name is confusing. The implementation in CompoundStatement returns the first ReturnStatement when backtracking through the tree, while with "endsWith" I'd expect it to be the last statement of a CompoundStatement.

Reply:
`hasReturnStatement`?

Reply:
Neither of the names actually make much sense when you think about its implementation, which also doesn't make much sense. Even worse, the astbase.d version does something different for `isReturnStatement()`.

Reply:
I guess the name is ok, but the comment could clarify what that means, i.e. it's a single return statement or some compound statement that unconditionally hits a return statement.

Reply:
ok done